### PR TITLE
Refactor declaration update and terminate functions using rpc

### PR DESCRIPTION
**Pull request by progsmile**
#4092

### CODE REVIEW

File: apps/core/lib/core/declarations/api.ex [hunk 1]
```diff
@@ -308,12 +330,21 @@ defmodule Core.Declarations.API do
     end
   end
 
+  def do_update_declaration(id, params) do
```
Comment (AlexKovalevych) [inline apps/core/lib/core/declarations/api.ex]: why do you need this function at all?

File: apps/core/lib/core/declarations/api.ex [hunk 2]
```diff
@@ -308,12 +330,21 @@ defmodule Core.Declarations.API do
...
+    with {:ok, declarations} <- @rpc_worker.run("ops", Core.Rpc, :update_declaration, [id, params]) do
+      {:ok, declarations}
+    end
+  end
+
+  def do_terminate_declaration(id, %{} = params) do
```
Comment (AlexKovalevych) [inline apps/core/lib/core/declarations/api.ex]: same question

File: apps/core/lib/core/jason_protocols.ex [hunk 3]
```diff
@@ -17,3 +17,11 @@ defimpl Jason.Encoder, for: Core.MedicationRequestRequest.EmbeddedData do
     Encode.map(Map.take(value, Core.MedicationRequestRequest.EmbeddedData.__schema__(:fields)), opts)
   end
 end
+
+defimpl Jason.Encoder, for: Core.Declarations.Declaration do
```
Comment (AlexKovalevych) [inline apps/core/lib/core/jason_protocols.ex]: why can't you render declarations at the ops service and receive a map with string keys in response? Cause as i understand, there is no module `Core.Declarations.Declaration` in the ehealth.api

File: apps/graphql/test/web/resolvers/declaration_resolver_test.exs [hunk 4]
```diff
@@ -416,6 +393,14 @@ defmodule GraphQLWeb.DeclarationResolverTest do
          }}
       end)
 
+      expect(RPCWorkerMock, :run, fn _, _, :get_declaration, _ -> declaration end)
+      expect(RPCWorkerMock, :run, fn _, _, :get_person_by_id, _ -> {:ok, person} end)
```
Comment (AlexKovalevych) [inline apps/graphql/test/web/resolvers/declaration_resolver_test.exs]: hm, how will it work, if the `:run` function is the same?

File: apps/core/lib/core/declarations/api.ex (hunk 1 above)
Comment (progsmile) [reply apps/core/lib/core/declarations/api.ex]: ok

File: apps/core/lib/core/declarations/api.ex (hunk 2 above)
Comment (progsmile) [reply apps/core/lib/core/declarations/api.ex]: ok

File: apps/graphql/test/web/resolvers/declaration_resolver_test.exs (hunk 4 above)
Comment (progsmile) [reply apps/graphql/test/web/resolvers/declaration_resolver_test.exs]: by order of mocks

File: apps/core/lib/core/jason_protocols.ex (hunk 3 above)
Comment (progsmile) [reply apps/core/lib/core/jason_protocols.ex]: Previously there was proxy response.. and we received declaration as string-keyed map. Now it's rpc call with structure, so may remove it in view

**Pull request merged**